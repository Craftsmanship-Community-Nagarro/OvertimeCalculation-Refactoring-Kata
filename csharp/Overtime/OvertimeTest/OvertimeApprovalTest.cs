using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using Overtime;
using System;
using System.Collections.Generic;
using Xunit;

namespace OvertimeTest
{
    [UseReporter(typeof(VisualStudioReporter))]
    public class OvertimeApprovalTest
    {
        [Fact]
        public void TestOvertimeCalculation()
        {
            var briefing = new Briefing(true, true, true, true);
            var assignment = new Assignment(true, TimeSpan.FromMinutes(1));

            var overtime = CompensationCalculator.calculateOvertime(1.0m, assignment, briefing);

            Approvals.Verify(overtime);
        }

        [Fact]
        public void TestOvertimeCalculationCombinations()
        {
            var hoursOvertimeTotals = new List<decimal>() { 1.0m, 3.5m, 6.8m };
            var briefings = new List<Briefing>() { 
                new Briefing(true, true, true, true), 
                new Briefing(false, true, true, true), 
                new Briefing(false, false, true, true) };
            var assignments = new List<Assignment>() {
                new Assignment(true, TimeSpan.FromMinutes(1)),
                new Assignment(true, TimeSpan.FromMinutes(2)),
                new Assignment(true, TimeSpan.FromMinutes(5)) };

            CombinationApprovals.VerifyAllCombinations(CompensationCalculator.calculateOvertime, hoursOvertimeTotals, assignments, briefings);
        }
    }
}