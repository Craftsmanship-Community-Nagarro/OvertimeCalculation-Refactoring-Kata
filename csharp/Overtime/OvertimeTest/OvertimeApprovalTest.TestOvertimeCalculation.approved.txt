Overtime -> hoursRate1:1,0_hoursRate2:0